osgar/drivers/oak_camera.py

```diff
@@ -137,22 +157,32 @@ def run_input(self):
 
         # Connect to device and start pipeline
         with dai.Device(pipeline, device_info) as device:
+            if self.laser_dot_projector:
+                device.setIrLaserDotProjectorBrightness(self.laser_dot_projector)
             while self.bus.is_alive():
                 queue_events = device.getQueueEvents(queue_names)
 
                 for queue_name in queue_events:
-                    if queue_name == "depth":
-                        packets = device.getOutputQueue(queue_name).tryGetAll()
-                        if len(packets) > 0:
+                    packets = device.getOutputQueue(queue_name).tryGetAll()
+                    if len(packets) > 0:
+                        if queue_name == "depth":
                             depth_frame = packets[-1].getFrame()  # use latest packet
                             self.bus.publish("depth", depth_frame)
 
-                    if queue_name == "color":
-                        packets = device.getOutputQueue(queue_name).tryGetAll()
-                        if len(packets) > 0:
+                        if queue_name == "color":
                             color_frame = packets[-1].getData().tobytes()  # use latest packet
                             self.bus.publish("color", color_frame)
 
+                        if queue_name == "rotation":
+                            for packet in packets:
+                                quaternions = [[data.rotationVector.getTimestampDevice().total_seconds(),  # timestamp
+                                                data.rotationVector.rotationVectorAccuracy,  # Accuracy in rad, zero for GAME_ROTATION_VECTOR ?
+                                                data.rotationVector.i, data.rotationVector.j,
+                                                data.rotationVector.k, data.rotationVector.real
+                                                ]
+                                               for data in packet.packets]
+                                self.bus.publish("rotation_list", quaternions)
 
     def request_stop(self):
+        self.bus.sleep(2)
```

Comment:
this is strange - maybe trigger the device termination?

Reply:
As we discussed, this is just a forgotten attempt. It seems that the problems with the process termination were related to a bad connection to the camera.